doc/design/api.md

```diff
@@ -0,0 +1,277 @@
+# PaddlePaddle Design Doc
+
+## Ingredients
+
+As the first step of our design, we list important concepts in deep
+learning and try to figure their relationship, as shown below:
+
+```
+Model = {topology, parameters}
+
+Evaluator = {Model*, activations}
+- forward
+- test(cost, ...)
```

Comment:
Even more aggressively, Is `Evaluator` a necessary concept for Paddle?

It seems if we think the activation should be the result of forwarding a neural network, `Evaluator` should store nothing. The global `forward` method is enough? or `forward` method should be a part of neural network model?